src/datasources/promise/promise-registry.spec.ts

```diff
@@ -12,32 +12,38 @@ describe('Promise Registry tests', () => {
   it('registration is successful', () => {
     const promise = Promise.resolve();
 
-    promiseRegistry.register('foo', () => promise);
+    promiseRegistry.register({ key: 'foo', field: 'bar' }, () => promise);
```

Comment:
After discussing this implementation with Fred, I've opted for a helper instead of converting the `key` of the `PromiseRegistry`. This keeps the `PromiseRegistry` more generic, should we want to use this in the future.

However, I have implemented your suggestion to the test.